lib/components/primitive-components/Trace/Trace.ts

```diff
@@ -781,12 +780,102 @@ export class Trace
       }),
     )
 
-    const trace = db.schematic_trace.insert({
-      source_trace_id: this.source_trace_id!,
-      edges,
-      junctions,
-    })
+    if (
+      this.props.schDisplayLabel &&
+      (("from" in this.props && "to" in this.props) || "path" in this.props)
```

Comment:
so for this logic i would implement an "early bail" style rather than letting the expensive autorouter run- just detect if the trace has a `schDisplayLabel` in the first couple lines then do...

```
if (schDisplayLabel) {
  this._doInitialSchematicTraceRenderWithDisplayLabel()
  return
}
```

these mega-functions get hard to read and it's also difficult to see how they get slow (in this case the autorouter running is totally unnecessary!)